source/host_resolver.c

```diff
@@ -131,15 +131,15 @@ struct host_entry {
 static int resolver_purge_cache(struct aws_host_resolver *resolver) {
     struct default_host_resolver *default_host_resolver = resolver->impl;
     aws_mutex_lock(&default_host_resolver->host_lock);
-    aws_lru_cache_clear(&default_host_resolver->host_table);
+    aws_cache_clear(default_host_resolver->host_table);
     aws_mutex_unlock(&default_host_resolver->host_lock);
 
     return AWS_OP_SUCCESS;
 }
 
 static void resolver_destroy(struct aws_host_resolver *resolver) {
     struct default_host_resolver *default_host_resolver = resolver->impl;
-    aws_lru_cache_clean_up(&default_host_resolver->host_table);
+    aws_cache_clean_up(default_host_resolver->host_table);
```

Comment:
should be `aws_cache_destroy()`
`_init()/_clean_up()` is for on-the-stack structs
`_new()/_destroy()` is for allocated-on-the-heap structs